lib/active_storage/service/s3_service.rb

```diff
@@ -4,8 +4,14 @@
 class ActiveStorage::Service::S3Service < ActiveStorage::Service
   attr_reader :client, :bucket
 
-  def initialize(access_key_id:, secret_access_key:, region:, bucket:)
-    @client = Aws::S3::Resource.new(access_key_id: access_key_id, secret_access_key: secret_access_key, region: region)
+  def initialize(access_key_id:, secret_access_key:, region:, bucket:, endpoint=nil)
```

Comment:
It should be optional, since most people who use the S3 service will actually be using S3. Set its default to `nil`.